pkg/build/api/v1/conversion.go

```diff
@@ -166,17 +166,28 @@ func convert_api_BuildSource_To_v1_BuildSource(in *newer.BuildSource, out *Build
 	if err := s.DefaultConvert(in, out, conversion.IgnoreMissingFields); err != nil {
 		return err
 	}
+	typeFound := false
 	switch {
 	// it is legal for a buildsource to have both a git+dockerfile source, but in v1 that was represented
 	// as type git.
 	case in.Git != nil:
 		out.Type = BuildSourceGit
+		typeFound = true
 	// it is legal for a buildsource to have both a binary+dockerfile source, but in v1 that was represented
 	// as type binary.
 	case in.Binary != nil:
 		out.Type = BuildSourceBinary
+		typeFound = true
 	case in.Dockerfile != nil:
 		out.Type = BuildSourceDockerfile
+		typeFound = true
+	}
+	if len(in.Images) > 0 && in.Git == nil && in.Binary == nil && in.Dockerfile == nil {
```

Comment:
for that matter I think it could move up into the switch, since go doesn't fallthrough by default?

`case len(in.Images)>0:`